flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTrackerImpl.java

```diff
@@ -120,6 +125,26 @@ public Collection<ResultPartitionDeploymentDescriptor> getAllTrackedPartitions()
         return partitionInfos.values().stream().map(PartitionInfo::getMetaInfo).collect(toList());
     }
 
+    @Override
+    public void connectToResourceManager(ResourceManagerGateway resourceManagerGateway) {
+        this.resourceManagerGateway = resourceManagerGateway;
+    }
+
+    @Override
+    public List<ShuffleDescriptor> getClusterPartitionShuffleDescriptors(
+            IntermediateDataSetID intermediateDataSetID) {
+        Preconditions.checkNotNull(
+                resourceManagerGateway, "JobMaster is not connected to ResourceManager");
+        try {
+            return this.resourceManagerGateway
+                    .getClusterPartitionsShuffleDescriptors(intermediateDataSetID)
+                    .get();
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
```

Comment:
`e.printStackTrace()` should be avoided.
Usually it should be a warning log with exception.
But I'm wondering why it is returning a fake result, instead of throwing out the exception to trigger failures?